atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -538,6 +545,19 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
         return transactionManager;
     }
 
+    private String getServiceName() {
+        String serviceName = "UNKNOWN";
+        if (config().namespace().isPresent()) {
+            serviceName = config().namespace().get();
+        } else if (config().timelock().isPresent()) {
+            TimeLockClientConfig timeLockClientConfig = config().timelock().get();
+            serviceName = timeLockClientConfig.getClientOrThrow();
+        } else if (config().keyValueService().namespace().isPresent()) {
+            serviceName = config().keyValueService().namespace().get();
+        }
+        return serviceName;
```

Comment:
Consider using a stream-of-optionals impl? It's probably going to be neater (provided it works)